### PR TITLE
add Contact Us page

### PR DESCRIPTION
**Pull request by theakinkavak**
Formik and emailJs still missing. I'm going to work on them. Mobile responsive for map will be developed.

-  

image

- 

image

- 

image

- 

image

### CODE REVIEW

File: src/pages/ContactUsPage.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import Header from '../components/ContactUs/Header';
+import Form from '../components/ContactUs/Form';
+import Address from '../components/ContactUs/Address';
+import Map from '../components/ContactUs/Map';
```
Comment (Iseluin) [inline src/pages/ContactUsPage.js]: Good work on the file structure!

File: src/components/ContactUs/Form.js [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+                      duration-150"
+        >
+          SEND
+        </button>
+      </form>
+    </div>
+  );
+}
+
+export default Form;
```
Comment (Iseluin) [inline src/components/ContactUs/Form.js]: Please re-do this with Formik library and update the PR. Since this is just the design development phase, let us not leave using Formik behind.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Good work on the file structure and page design Akın! But let's please use Formik and EmailJs so this will be a complete PR. EmailJs is a very lightweight library and I believe you can do it easily.

**Comment by Iseluin** [comment]
The resolution of the illustration we have next to our form looks a bit off. Can you check its resolution? If you imported it from Figma, please try to import it as SVG so it can keep its resolution.

The image also has responsivity issues, it distorts and does not response proportionally to screen size changes. See the added screenshot for reference. Please change your code so that the image does not distort on smaller screens.

Plus, on smaller sized screens, the illustration should be larger like we had in our Figma design. Please alter that.

Responsivity issue

Formik and emailjs parts are still WIP so I will be reviewing them after. Lastly, please do not forget to update your branch from dev branch since we had some merges.

**Comment by theakinkavak** [comment]
- Set validation and error messages using Formik.
- Create a function to acces gmail service when the form is submitted using EmailJs.
- Fix resolution of the illustration using SVG format with @bayogluteoman.
- Refactor labels to inputs' elements.

File: .eslintrc.json [hunk 3]
```diff
@@ -28,6 +28,7 @@
         "react/jsx-props-no-spreading":["warn"],
         "no-param-reassign":["off"],
         "react/jsx-filename-extension": [1, { "extensions": [".js", ".jsx"] }],
-        "testing-library/prefer-screen-queries":["off"]
+        "testing-library/prefer-screen-queries":["off"],
+        "no-alert":"off"
```
Comment (Iseluin) [inline .eslintrc.json]: Why did you add this no-alert exception ?

File: src/components/ContactUs/Form.js [hunk 4]
```diff
@@ -11,7 +11,7 @@ function Form() {
     emailjs
```
Comment (Iseluin) [inline src/components/ContactUs/Form.js]: This emailing function seems nice, so it can live in another file that we will set up for our helper functions. Can you try to move the function to a file called utils.js inside a utils folder. That way we can import it and use it so the code will look much cleaner.

File: src/components/ContactUs/Form.js [hunk 5]
```diff
@@ -11,7 +11,7 @@ function Form() {
...
       .sendForm(
         'service_qozhrg7',
-        'emplate_akxu7oa',
+        'template_akxu7oa',
         form.current,
         'B2JkePBOzV_nhSmeO',
       )
```
Comment (Iseluin) [inline src/components/ContactUs/Form.js]: Also, in the function, when the emailing request is succesfull, it gives back "OK" as an alert. It does not look good at all. Can you change the alert to a notification? 

You can have a look at this library: https://www.npmjs.com/package/react-notifications

Other than that, let us show something relevant on that notification other than just "OK" for a better user experience.

**Review by Iseluin — CHANGES_REQUESTED** [review]
Great work Akın. We have just a bit more enhancements to do and we will be good to go. Apart from the comments I left down below, please be sure to complete these 3 points too :

- Let's add unit and snapshot tests
- I cannot confirm if the email functionality works other than the OK alert we are having, so can you check if my test mail is in your inbox ? 
- Let us no forget to update our local branch and resolve the merge conflicts before the merge.

**Comment by haticecs** [comment]
In the **authentication pages error message in the form is italic but it is normal here. We may decide one of the text style thus, it seems more professional**. Apart from that, it looks so good. @theakinkavak @orkunseyman

**Comment by Iseluin** [comment]
> In the **authentication pages error message in the form is italic but it is normal here. We may decide one of the text style thus, it seems more professional**. Apart from that, it looks so good. @theakinkavak @orkunseyman

Good eye Hatice! We can fix the font styles in the bug bashing part of the project so it's not a big issue for now. The pressing matter is the tests.

File: src/components/ContactUs/Form.js [hunk 6]
```diff
@@ -17,10 +18,12 @@ function Form() {
...
       .then(
         (result) => {
-          alert(result.text);
+          <Notification data={result} />;
+          // eslint-disable-next-line no-console
+          console.log('notification is sent!');
```
Comment (Iseluin) [inline src/components/ContactUs/Form.js]: Please remove the console.log and the eslint exception. I believe you added the console log just for testing but the problem is, I cant see the notification after I submit my message, I only can see the console.log.

File: src/components/ContactUs/Notification.js [hunk 7]
```diff
@@ -0,0 +1,83 @@
+// eslint-disable-next-line react/prop-types
```
Comment (Iseluin) [inline src/components/ContactUs/Notification.js]: No exceptions please, you can add prop types to your props so you  wont have to add an exception like this. May google be on your side

**Review by Iseluin — CHANGES_REQUESTED** [review]
Hey Akın, I see what you did here but there seems to be a problem, the notification modal is not showing when clicked on submit. Fix that please, and check the other minor things I saw in your code below.

File: src/components/ContactUs/Form/Form.js [hunk 8]
```diff
@@ -0,0 +1,193 @@
+import { Formik } from 'formik';
+import { useRef } from 'react';
+import emailjs from '@emailjs/browser';
+import { ToastContainer, toast } from 'react-toastify';
```
Comment (Iseluin) [inline src/components/ContactUs/Form/Form.js]: A blank line before the local imports please.

File: src/components/ContactUs/Form/Form.js [hunk 9]
```diff
@@ -0,0 +1,193 @@
...
+    toast.success('Your messsage has been sent successfully!', {
+      position: 'top-right',
+      autoClose: 5000,
+      hideProgressBar: false,
+      closeOnClick: true,
+      pauseOnHover: true,
+      draggable: true,
+      progress: undefined,
+    });
+  };
```
Comment (Iseluin) [inline src/components/ContactUs/Form/Form.js:29]: please move this function to utils/helpers.js as a helper function, import it in here and then use it. It will enhance the code readabiity.

File: src/components/ContactUs/Map/Map.js [hunk 10]
```diff
@@ -0,0 +1,43 @@
...
+      'pk.eyJ1IjoidGhlYWtpbmthdmFrIiwiYSI6ImNsMWwzY21qYjAxdDczYnBxMDllY2k3YXEifQ.UPpP7OQtEb-wfVWnxqxV0A',
+  });
+
+  return (
+    <div
+      className="flex text-center lg:text-center items-center justify-center"
+      data-testid="valid-map"
+    >
+      <MapDetail
+        // eslint-disable-next-line react/style-prop-object
```
Comment (Iseluin) [inline src/components/ContactUs/Map/Map.js:21]: Does't this work without this exception ? We don't add exceptions.

File: src/pages/ContactUs.jsx [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+        <div className="flex justify-center lg:justify-center">
+          <Form />
+        </div>
+
+        <div className="flex justify-center">
+          <Image className="p-1 w-3/4" alt="contact-img" />
+        </div>
+
+        <Map />
+
```
Comment (Iseluin) [inline src/pages/ContactUs.jsx]: These blank lines might create some prettier issues, please delete them.

**Review by Iseluin — CHANGES_REQUESTED** [review]
1. Please move the images to a folder called `src/assets/images`
2. We can ommit writing tests for `Map` component.

File: .eslintrc.json (hunk 3 above)
Comment (theakinkavak) [reply .eslintrc.json]: To prevent eslint warning to check other component's functionality. Forgot to remove it but now just removed and fixed it.

File: src/components/ContactUs/Form.js (hunk 6 above)
Comment (theakinkavak) [reply src/components/ContactUs/Form.js]: The console log and eslint exception were removed. In addition, I installed `react-toastify` library to get notification when the form is submitted. 
- 
toastify

File: src/components/ContactUs/Notification.js (hunk 7 above)
Comment (theakinkavak) [reply src/components/ContactUs/Notification.js]: Done. There will be no more eslint exceptions anymore =)

File: src/components/ContactUs/Form/Form.js (hunk 9 above)
Comment (theakinkavak) [reply src/components/ContactUs/Form/Form.js:29]: Should i create `utils` folder inside `src/Form/` or `src/`?

File: src/components/ContactUs/Map/Map.js (hunk 10 above)
Comment (theakinkavak) [reply src/components/ContactUs/Map/Map.js:21]: When I try to compile the next line, I get the following warning: **style prop value must be an object.** When I changed it to an object, the map is not rendering.

File: src/components/ContactUs/Form/Form.js (hunk 9 above)
Comment (Iseluin) [reply src/components/ContactUs/Form/Form.js:29]: `src/utils/helpers.js`

**Comment by OzerOzturk** [comment]
Frankly, it seems like a bit of hard work, but you did a good job @theakinkavak by breaking it down into small pieces and improving it.

**Comment by Iseluin** [comment]
The notification sends out a success notice even if we dont write the required fields. Please fix that.
